graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/DmlFetcher.java

```diff
@@ -109,24 +110,64 @@ protected DmlFetcher(Table table, NameMapping nameMapping) {
     }
   }
 
-  protected List<Map<String, Object>> toListOfMutationResults(
-      ResultSet resultSet, List<Map<String, Object>> originalValues) {
-    List<Row> rows = resultSet.currentPageRows();
+  protected List<Map<String, Object>> toBatchResults(
+      List<Row> rows, List<Map<String, Object>> originalValues) {
+
+    boolean applied = isAppliedBatch(rows);
+
     List<Map<String, Object>> results = new ArrayList<>();
-    if (rows.isEmpty()) {
-      for (Map<String, Object> originalValue : originalValues) {
-        results.add(toAppliedMutationResultWithOriginalValue(originalValue));
-      }
-    } else {
-      for (int i = 0; i <= rows.size(); i++) {
-        Row row = rows.get(i);
-        Map<String, Object> originalValue = originalValues.get(i);
-        results.add(toMutationResultSingleRow(originalValue, row));
-      }
+    for (Map<String, Object> originalValue : originalValues) {
+      results.add(
+          applied
+              ? toAppliedMutationResultWithOriginalValue(originalValue)
+              : toUnappliedBatchResult(rows, originalValue));
     }
     return results;
   }
 
+  protected Map<String, Object> toBatchResult(List<Row> rows, Map<String, Object> originalValue) {
+    return isAppliedBatch(rows)
+        ? toAppliedMutationResultWithOriginalValue(originalValue)
+        : toUnappliedBatchResult(rows, originalValue);
+  }
+
+  private boolean isAppliedBatch(List<Row> batchRows) {
+    if (batchRows.isEmpty()) {
```

Comment:
It means that if there are no rows returned we treat it as success. However, also if one row is returned we are extracting it's applied. Could you elaborate on when the first and the second situation are happening?

Reply:
No rows is when a batch has no conditions:
```
cqlsh:test> begin batch
        ... insert into foo(k,cc,v) values(1,1,1);
        ... insert into foo(k,cc,v) values(1,2,2);
        ... apply batch;

<< no rows (void) >>
```
One row with `applied` is when the batch has conditions that are met:
```
cqlsh:test> truncate table foo;
cqlsh:test> begin batch
        ... insert into foo(k,cc,v) values(1,1,1) if not exists;
        ... insert into foo(k,cc,v) values(1,2,2);
        ... apply batch;

 [applied]
-----------
      True
```